### PR TITLE
DROOLS-4119: [DMN Designer] Documentation tab - Show information about each DRD component in the documentation tab

### PR DESCRIPTION
**Pull request by karreiro**
See:
- https://issues.jboss.org/browse/DROOLS-4119
- https://issues.jboss.org/browse/DROOLS-4199

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
@jomarko @danielzhe Could you please take a look? :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        final View content = mock(View.class);
+        final InputData expectedHasExpression = mock(InputData.class);
+
+        when(node.getContent()).thenReturn(content);
+        when(content.getDefinition()).thenReturn(expectedHasExpression);
+
+        final Optional<HasExpression> actualHasExpression = helper.getOptionalHasExpression(node);
+
+        assertFalse(actualHasExpression.isPresent());
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java]: According this test, is it OK that InputData is in documentation?
input-data

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    @Test
+    public void testGetOptionalExpressionWhenIsPresent() {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java]: I think `testGetOptionalExpressionWhenIsPresent` and `testGetOptionalExpressionWhenIsNotPresent` are swapped.

Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java]: Thank you, @jomarko. Fixed!

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/common/BoxedExpressionHelperTest.java]: Yes, @jomarko. We're showing all nodes from the graph in into documentation.

**Comment by karreiro** [comment]
Thank you for the review, @jomarko. I've applied the fix in a separated commit.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Pull request merged**